soc: nxp_imx: rt1064: Ethernet bringup

loicpoulain opened the pull request:
Enable on-board flash (4MB) (flexspi2).
Enable ethernet support.
Fix missing 4MB ARM/MPU REGION_FLASH_SIZE.

Signed-off-by: Loic Poulain <loic.poulain@linaro.org>

Commit:
soc: nxp_imx: Add on-chip flash code location support

iMXRT1064 has 4MB on-chip Flash memory (flexspi2) mapped to 0x70000000.
Add it as configurable code location for MIMXRT1064-EVK board.

Signed-off-by: Loic Poulain <loic.poulain@linaro.org>

MaureenHelm commented on boards/arm/mimxrt1064_evk/Kconfig.defconfig:
```diff
@@ -10,6 +10,14 @@ if BOARD_MIMXRT1064_EVK
 	default "mimxrt1064_evk"
 
+choice CODE_LOCATION
+        default CODE_FLEXSPI2
```
How are you programming the code to flash? Last I checked, JLink didn't support this yet.

@jhqian Did Segger resolve this issue recently?

loicpoulain replied:
> How are you programming the code to flash? Last I checked, JLink didn't support this yet.

I use drag-and-drop method for now.

MaureenHelm replied:
Let's keep the defaults (both code and data) in TCM until the tools can properly support flash programming.

MaureenHelm commented on boards/arm/mimxrt1064_evk/mimxrt1064_evk_defconfig:
```diff
@@ -14,3 +14,4 @@ CONFIG_SERIAL=y
 CONFIG_CORTEX_M_SYSTICK=y
 CONFIG_GPIO=y
 CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=600000000
+CONFIG_ARM_MPU=y
```
This change isn't needed after commit ee3f7d629b9d3f2fae06351bfa0e2dfece1cf935

MaureenHelm commented on soc/arm/nxp_imx/rt/Kconfig.soc:
```diff
@@ -99,14 +99,15 @@ config SOC_MIMXRT1062
 	select HAS_MCUX_TRNG
 	select CPU_HAS_FPU_DOUBLE_PRECISION
-	select CPU_HAS_MPU
+	select CPU_HAS_ARM_MPU
```
This change isn't needed after commit ee3f7d629b9d3f2fae06351bfa0e2dfece1cf935

loicpoulain replied:
ok thanks

MaureenHelm commented on soc/arm/nxp_imx/rt/arm_mpu_mem_cfg.h:
```diff
@@ -22,6 +22,8 @@
 #elif CONFIG_FLASH_SIZE == 2048
 #define REGION_FLASH_SIZE REGION_2M
+#elif CONFIG_FLASH_SIZE == 4096
+#define REGION_FLASH_SIZE REGION_4M
```
Move this change to the previous commit that added support for the internal flash.

loicpoulain replied:
ok

Commit:
soc: nxp_imx: mimxrt1064_evk: Enable ethernet support

The i.MX RT1064 evk has one ethernet (10/100M) connector via KSZ8081RNB
phy. Enable related dts nodes and config flags

Signed-off-by: Loic Poulain <loic.poulain@linaro.org>

MaureenHelm reviewed (changes requested):
Thanks for updating the docs and board yaml along with the code change!

MaureenHelm commented:
Thanks @loicpoulain. I see you changed the code default to ITCM, but will you also please change the data default to DTCM?

loicpoulain commented:
> Thanks @loicpoulain. I see you changed the code default to ITCM, but will you also please change the data default to DTCM?

Yes done, thx.
